design/ceph/ceph-mon-pv.md

```diff
@@ -136,3 +136,132 @@ relies on determining if monitor pods may be placed on a node, requiring
 knowledge of the scheduling policy and algorithm. The scenario to watch out for
 is an endless loop in which the health check is determining a bad placement but
 the k8s schedule thinks otherwise.
+
+## Monitor Storage Class Support
+
+It is generally desirable in production to have multiple zones availability. A
+whole storage class could go unavailable then and enough mons could stay up to
+keep the cluster running.
+In such a case the distribution of mons can be made more homogeneous for each
+zone, by this we can leverage the spread of mons for providing high
+availability. This could be a useful scenario for a datacenter with own storage
+class in contrast to ones provided by cloud providers.
+
+Rook will enable the mon deployment when requested by the admin through the
+usual Rook CRDs.
+
+To support this scenario, we can do something similar to the mon deployment for
+[Stretch clusters](https://github.com/ceph/ceph/blob/master/doc/rados/operations/stretch-mode.rst).
+The operator will have zonal awareness, associating mon deployments based on
+zones configuration provided.
+
+The type of failure domain used for this scenario is commonly "zone", but can
+be set to a different failure domain.
+
+### Failure Domain Configuration
+
+The topology of the K8s cluster is to be determined by the admin, outside the scope of Rook.
+Rook will simply detect the topology labels that have been added to the nodes.
+
+If the desired failure domain is a "zone", the `topology.kubernetes.io/zone` label should
+be added to the nodes. Any of the [topology labels](https://rook.io/docs/rook/latest/ceph-cluster-crd.html#osd-topology)
+supported by OSDs can be used.
+
+For example:
+
+```yaml
+topology.kubernetes.io/zone=a
+topology.kubernetes.io/zone=b
+topology.kubernetes.io/zone=c
+```
+
+###  Design
+
+The core changes to rook are to associate the mons with the required zones.
+
+- Each zone will have at least one mon assigned using labels for nodeAffinity.
+
+The new configuration will support the `zones` configuration where the
+failure domains available must be listed.
+
+```yaml
+  mon:
+    count: 3
+    allowMultiplePerNode: false
+    zones:
+      - name: us-east-2a
+      - name: us-east-2b
+      - name: us-east-2c
+    volumeClaimTemplate:
```

Comment:
The scenario really requires a volumeClaimTemplate to be specified for each zone. So how about we replace this example with the one later in the doc. One example seems sufficient.